privacyidea/lib/smsprovider/FirebaseProvider.py

```diff
@@ -73,12 +80,17 @@ def submit_message(self, firebase_token, data):
                 from_json_keyfile_name(self.smsgateway.option_dict.get(FIREBASE_CONFIG.JSON_CONFIG),
                                        SCOPES)
 
-        access_token_info = credentials.get_access_token()
-
-        # Should we do something with expires in?
-        # expires_in = access_token_info.expires_in
-
-        bearer_token = access_token_info.access_token
+        now = time.time()
+        if self.access_token_info is None or now > self.access_token_expires_at:
```

Comment:
I think this doesn't hurt, but I'm not sure if it will have an effect: Caching the access token would only speed up things if we call the ``submit_message`` method multiple times on the same ``FirebaseProvider`` object. As the ``FirebaseProvider`` objects do not persist between requests, this would only help if we call ``submit_message`` multiple times during the same request. I'm not sure if we do that?

Reply:
To bad the object gets destroyed. No, to my knowledge we do not call several submit_messages in one requst.

So how should we persist the access_token. Does only look like, as if we could persist it in the database. But to me it feels a bit awkward writing it to the SQL database.

Reply:
done, check it out :-)